### PR TITLE
Add Silver-Mueller boundaries along z, for the RZ version of the code

### PR DESCRIPTION
**Pull request by RemiLehe**
This builds on #1804 (which introduced Silver-Mueller boundaries in RZ, in the radial direction) and adds the code for the Silver-Mueller boundary conditions in z.

I also added an automated test which simulates the propagation and absorption of a laser into the z boundary conditions:
movie

### CODE REVIEW

**Review by ax3l — APPROVED** [review]
Thank you, looks great!

Waiting for @LDAmorim for co-review :)

File: Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp [hunk 1]
```diff
@@ -57,23 +60,50 @@ void FiniteDifferenceSolver::ApplySilverMuellerBoundary (
...
+        tbr.grow(1);
         tbt.grow(1);
         tbz.grow(1);
 
         // Loop over the cells
-        amrex::ParallelFor(tbt, tbz,
+        amrex::ParallelFor(tbr, tbt, tbz,
             [=] AMREX_GPU_DEVICE (int i, int j, int /*k*/){
 
+                // At the +z boundary (innermost guard cell)
```
Comment (LDAmorim) [inline Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp:83]: Hi Rémi, Thanks for asking me to review this PR.
Everything in the code and test looks very good to me.

Are you focusing only on the +z boundary for the moving window cases?

To help with the review, I tried to do the calculations (also to expand the PDF and update it for others to use it), but I got different equations although the coefficients all match yours.
Do you know where my deduction is different than yours?

[SilverMueller.pdf](https://github.com/ECP-WarpX/WarpX/files/6350166/SilverMueller.pdf)

Thanks

**Comment by RemiLehe** [comment]
Thanks a lot for checking the equations, @LDAmorim !
Your document is absolutely correct. In this PR, I neglected some of the terms that appear in your document, in order to implement this faster (since neglecting these terms still allowed to absorb the laser in normal incidence). More specifically, I neglected the transverse derivatives (the terms that look like `im/r`, and the ones that look like derivatives with respect to `r`). I actually did the same kind of approximation in the current 3D Cartesian version of the code, where I neglected some of the transverse derivatives. 

These more accurate terms could certainly be added as a follow-up PR.

**Comment by LDAmorim** [comment]
Hi Rémi, thanks for explaining it! Everything is perfect and works well. So I'll approve now. Thanks!

**Review by LDAmorim — APPROVED** [review]

**Comment by RemiLehe** [comment]
Thanks @LDAmorim and @ax3l for reviewing the code!

**Pull request merged**